File: work-spring/BoardOAuth/src/main/java/com/grepp/boardjwt/model/dto/TodoDTO.java
package com.grepp.boardjwt.model.dto;


import com.grepp.boardjwt.model.entity.TodoEntity;

public class TodoDTO {
  private int id;
  private String title;
  private boolean done;

  public TodoDTO(){}

  public TodoDTO(final TodoEntity entity) {
    this.id = entity.getId();
    this.title = entity.getTitle();
    this.done = entity.isDone();
  }

  public TodoEntity toEntity() {
    TodoEntity entity = new TodoEntity();
    entity.setId(this.id);
    entity.setDone(this.done);
    entity.setTitle(this.title);
    return entity;
  }

  public int getId() {
    return id;
  }

  public void setId(int id) {
    this.id = id;
  }

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public boolean isDone() {
    return done;
  }

  public void setDone(boolean done) {
    this.done = done;
  }

  @Override
  public String toString() {
    return "TodoDTO{" +
            "id=" + id +
            ", title='" + title + '\'' +
            ", done=" + done +
            '}';
  }
}

File: work-kotlin/day1021_todo/src/main/kotlin/com/grepp/todo/model/dto/TodoDTO.kt
package com.grepp.todo.model.dto

import com.grepp.todo.model.entity.TodoEntity

data class TodoDTO(
    var id:Int,
    var title:String,
    var done:Boolean,
) {
    // entity는 디비에 넣는 todo니까 사용자 누구인지 기록해야 됐었음.
    // dto는 현재 로그인한 사용자가 자기 todo 목록 보는건데 굳이 거기에 username을 기록할 필요가 있음?
    fun toEntity():TodoEntity{
        return TodoEntity(id=id, title=title,done=done)
    }
}